stable/kong/values.yaml

```diff
@@ -283,6 +283,7 @@ env:
   admin_error_log: /dev/stderr
   admin_gui_error_log: /dev/stderr
   portal_api_error_log: /dev/stderr
+  prefix: /kong_prefix/
```

Comment:
This is kind of weird.
Software packages don't usually create a directory under `/`.
The practice I've seen most widely used is to create such a working directory in `/opt`.

I'm not sure but there might be some security policies to stop processes from writing to `/`.

Thoughts?

(Of course, Kong should not log to that but under `/var/log` but that's a bigger conversation for another time.)

Reply:
We live in container land and traditional Unix hierarchy no longer matters :D

The choice was essentially arbitrary, but based on the existing mounts at `/kong` and `/kong_dbless` at https://github.com/helm/charts/blob/master/stable/kong/templates/_helpers.tpl#L179-L185

As far as I know, the only Kubernetes-level restrictions are on allowing root writes (at the mount level) and what types of volumes can be written to. The second part probably needs further review: the existing PSP template [does not allow emptyDir writes](https://github.com/helm/charts/blob/master/stable/kong/templates/psp.yaml#L23-L25), which should break this. However, I didn't encounter issues when enabling the PSP, which is curious.

Standard FS permissions apply within the container, but anything at the root gets its own set of permissions (i.e. there shouldn't ever be any issue traversing to it for lack of execute on an intermediate directory), and emptyDirs get mounted with all access to everyone:

```
ls -ld /kong_prefix/
drwxrwxrwx 11 root root 4096 Dec 10 19:38 /kong_prefix/
[kong@scared-koala-kong-d7f8748f5-ml9jp /]$ env | grep PATH
KONG_LUA_PACKAGE_PATH=/opt/?.lua;;
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```
Part of the goal of the read-only root configuration is to avoid having writeable locations in the PATH, to guard against an attacker writing and then invoking malicious code or binaries. Our usage of `/opt` in the Lua path should preclude placing the prefix under there as such, though we separately need to make those ConfigMap mounts read-only also for that to work.